content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1441 @@
+
+```html
+<body>
+    <!-- Step 2 code additions - start -->
+    <h3>Build a XRPL Wallet - Part 2/8</h3>
+    <b>Latest validated ledger stats</b><br />
+    Ledger Index: <strong id="ledger-index"></strong><br />
+    Ledger Hash: <strong id="ledger-hash"></strong><br />
+    Close Time: <strong id="ledger-close-time"></strong><br />
+    <!-- Step 2 code additions - end -->
+</body>
+```
+
+5. Modify the `view/renderer.js` file to handle the new placeholders. Note that the renamed preloader function is now reflected in `window.electronAPI.onUpdateLedgerData`:
+
+```javascript
+const ledgerIndexEl = document.getElementById('ledger-index')
+const ledgerHashEl = document.getElementById('ledger-hash')
+const ledgerCloseTimeEl = document.getElementById('ledger-close-time')
+
+window.electronAPI.onUpdateLedgerData((_event, value) => {
+    ledgerIndexEl.innerText = value.ledger_index
+    ledgerHashEl.innerText = value.ledger_hash
+    ledgerCloseTimeEl.innerText = value.ledger_time
+})
+```
+
+This should make our application listen to regular updates of the ledger and display them in the frontend.
+
+Now run the application with the following command:
+
+```console
+npm run start
+```
+
+This time the application should be the same as in the last step, with the difference that the latest ledger index value gets updated roughly very 3-5 seconds.
+
+To run the reference application found in `content/_code-samples/build-a-wallet/desktop-js` for this step, run:
+
+```console
+npm run async
+```
+
+### 3. Display an Account
+
+**Full code for this step:** 
+[`3_account.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/3_account.js).
+[`library/3_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/3_helper.js).
+[`view/3_preload.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/3_preload.js).
+[`view/3_account.html`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/3_account.html).
+[`view/3_renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/3_renderer.js).
+
+We now have a permanent connection to the XRPL and some code to bring the delivered data to life on our screen, it's time to add some "wallet" functionality by managing an individual account. 
+
+We will ask the user for address of the account to monitor by using a HTML dialog element. We will furthermore refactor the application by encapsulating some functionality in a library. After finishing this step the application should look like this:
+
+![Screenshot: Step 3, show account information](img/javascript-wallet-3.png)
+
+1. In the project root, create a new directory named `library`. Inside this directory, create a file `3_helpers.js` with the following content:
```

Comment:
This file should probably just be called `helpers.js` instead of `3_helpers.js` for the actual tutorial follow-along parts.

Reply:
From Step 3 on on each iteration there will be an additional helper file in order to build kind of a library. I'll leave this open for now, perhaps at the end of the tutorial it makes sense or we can come up with something better than one very large file.

Reply:
Makes sense :)

Reply:
Then I can close this :)